### ISSUE TITLE
Request for providing additional options

### ISSUE THREAD

**Issue by sander48k** [PlayerData/react-native-mcu-manager#33]
To improve firmware upload speed and reliability I propose to add the following parameters to UpgradeOptions:

- Window capacity/pipeline depth (number); currently, the default value is 1. a higher value might increase the upload speed.
- Byte alignment; currently, the default value of 1 (disabled) is used. Setting this to a value that corresponds to the byte alignment of the device might increase speed.
- Request high connection priority (boolean); The Android 'Nordic Device Manager' App requests a high connection priority before starting the upgrade process. See [here](https://github.com/NordicSemiconductor/Android-nRF-Connect-Device-Manager/blob/main/sample/src/main/java/io/runtime/mcumgr/sample/viewmodel/mcumgr/ImageUpgradeViewModel.java#L170).

**Comment by ball-hayden** [comment]
This would be great to see @sander48k.
We do have an item on our backlog to upgrade the underlying libraries so that we can expose these, but we haven't got to it yet.

I did some initial work in https://github.com/PlayerData/react-native-mcu-manager/pull/29 - I'd be happy if you'd like to take over with that?

**Comment by sander48k** [comment]
@ball-hayden ; thanks for the Swift reply! I'll check if I can make some time to implement these changes.

**Comment by sander48k** [comment]
Hi @ball-hayden ; I've been working on this a little this weekend. Now I see that my code changes are almost the same as yours in the branch you mention in #29, which is a good thing :). Only difference is that I made an enum for the byte alignment because that is restricted in iOS.

My first test shows a significant speed improvement on Android (which you also saw already). On iOS, using the latest version 1.2.9, it also seems to work fine although I haven't tested it thoroughly yet. I find it remarkable that it is required to start the update process on the main thread. Do you have any idea what the rationale behind that is?

**Comment by ball-hayden** [comment]
No. There are many things that confuse me about Nordic's implementation.

**Comment by sander48k** [comment]
The improvements to the update process are in [this branch](https://github.com/playpicoo/react-native-mcu-manager/tree/feature/additional_upgrade_options). It would be nice if the new upgrade options are available in the example project but I'm afraid I don't have time for that now. I'll create a PR if you want.

**Comment by ball-hayden** [comment]
Thanks. Please do open a PR - I'd be happy to review that.
